util/network-devp2p/src/discovery.rs

```diff
@@ -419,10 +412,10 @@ impl<'a> Discovery<'a> {
 		Ok(())
 	}
 
-	fn send_packet(&mut self, packet_id: u8, address: &SocketAddr, payload: &[u8]) -> Result<H256, Error> {
+	fn send_packet(&mut self, packet_id: u8, address: SocketAddr, payload: &[u8]) -> Result<H256, Error> {
 		let packet = assemble_packet(packet_id, payload, &self.secret)?;
 		let hash = H256::from_slice(&packet[0..32]);
```

Comment:
This is a bit annoying, we just built one of these in `assemble_packet()`; maybe return the `signed_hash` from there?

Reply:
hehe, right I do get your concern

In terms, of `clean code` if we change `assemble_packet` then it does clearly more than one thing but it could be quite nice if we could introduce another type `Packet (or similar)` to avoid `copy_from_slice` stuff that would quite nice.

But, let's address it another PR :)

Reply:
Do you really think having `assemble_packet` return `Result<(Bytes, H256), Error>` would be so bad? It's a private method after all, and it wouldn't be doing more or different work?

> But, let's address it another PR :)

Absolutely.

Reply:
>Do you really think having assemble_packet return Result<(Bytes, H256), Error> would be so bad? It's a private method after all, and it wouldn't be doing more or different work?

It is fine but my general opinion it should not :)